src/pages/embed/[uid].tsx

```diff
@@ -32,25 +42,39 @@ const BrowserEmbedViewCore: React.FC<{ userId: string }> = ({ userId }) => {
   );
 };
 
-const BrowserEmbedView: React.FC<{ userId: string }> = (props) => {
+const LazyEmbedView = dynamic(() => Promise.resolve(BrowserEmbedViewCore), {
+  ssr: false,
+});
+
+const BrowserEmbedView: React.FC<ServerSideProps> = (props) => {
+  if (!props.userId) return null;
+
   return (
     <PusherProvider slug={`user-${props.userId}`}>
-      <BrowserEmbedViewCore {...props} />
+      <LazyEmbedView {...props} />
     </PusherProvider>
   );
 };
 
-const LazyEmbedView = dynamic(() => Promise.resolve(BrowserEmbedView), {
-  ssr: false,
-});
+export default BrowserEmbedView;
 
-const BrowserEmbedQuestionView = () => {
-  const { query } = useRouter();
-  if (!query.uid || typeof query.uid !== "string") {
-    return null;
-  }
+export const getServerSideProps = async (
+  context: GetServerSidePropsContext
+) => {
+  const uid = context.query.uid;
 
-  return <LazyEmbedView userId={query.uid} />;
-};
+  if (typeof uid !== "string") return { props: { success: false } };
 
-export default BrowserEmbedQuestionView;
+  const pinnedQuestion = await prisma.question
+    .findFirst({
```

Comment:
I feel like this call but not be very efficient if we get a lot of questions in the DB (which is likely)

Reply:
Resulting in a slight delay in page load time, vs the current.... not getting anything until the user pins a new question.

I think this is a non-issue. Not to mention that SQL is kind of really fast. It's not like we're working with millions of rows here.